frb_codegen/src/generator/dart/mod.rs

```diff
@@ -221,48 +261,156 @@ fn generate_dart_declaration_body(
+                {plat}(ffi.DynamicLibrary dylib) : super({wire}(dylib));",
+            plat = dart_platform_class_name,
+            wire = dart_wire_class_name,
+        ),
+        Wasm => format!(
+            "class {plat} extends FlutterRustBridgeBase<{wire}> with FlutterRustBridgeSetupMixin {{
+                {plat}(FutureOr<WasmModule> dylib) : super({wire}(dylib)) {{
+                    setupMixinConstructor();
+                }}
+                Future<void> setup() => inner.init;",
+            plat = dart_platform_class_name,
+            wire = dart_wire_class_name,
+        ),
+    }));
 
-            // Section: api_fill_to_wire
-            {}
-        }}
+    lines.extend(dart_funcs.iter().map(|func| {
+        format!(
+            "{}\n\n{}",
+            func.implementation, func.companion_field_implementation,
+        )
+    }));
+
+    lines.push("}\n".into());
+
+    lines.push_all(section_header("api2wire"));
+    lines.push_acc(dart_api2wire_funcs.clone());
+
+    lines.io.push(section_header("api_fill_to_wire"));
+    lines.io.push(dart_api_fill_to_wire_funcs.join("\n\n"));
+
+    lines.push_acc(Acc::new(|target| match target {
+        Io | Wasm => "}\n".into(),
+        Common => "".into(),
+    }));
+
+    lines.push(section_header("wire2api"));
+    lines.push(dart_wire2api_funcs.join("\n\n"));
+
+    if config.wasm_enabled {
+        let dart_wasm_module_name = config.dart_wasm_module();
+        lines.wasm.push(section_header("WASM wire module"));
+        lines.wasm.push(format!(
+            "@JS('wasm_bindgen') external {} get wasmModule;",
+            dart_wasm_module_name,
+        ));
+        lines.wasm.push(format!(
+            "@JS() @anonymous class {wasm} implements WasmModule {{
+                external Object /* Promise */ call([String? moduleName]);
+                external {wasm} bind(dynamic thisArg, String moduleName);
+            ",
+            wasm = dart_wasm_module_name,
+        ));
+        lines.wasm.push(dart_wasm_funcs.join("\n\n"));
+        lines.wasm.push("}\n".into());
+
+        lines.wasm.push(section_header("WASM wire connector"));
+        (lines.wasm).push(dart_wasm_module.as_deref().unwrap_or_default().to_string());
+    }
 
-        // Section: wire2api
-        {}
-        ",
-        dart_funcs
-            .iter()
-            .map(|func| format!(
-                "{}\n\n{}",
-                func.implementation, func.companion_field_implementation,
-            ))
-            .collect::<Vec<_>>()
-            .join("\n\n"),
-        dart_api2wire_funcs.join("\n\n"),
-        dart_api_fill_to_wire_funcs.join("\n\n"),
-        dart_wire2api_funcs.join("\n\n"),
-        dart_api_impl_class_name = dart_api_impl_class_name,
-        dart_wire_class_name = dart_wire_class_name,
-        dart_api_class_name = dart_api_class_name,
-    )
+    let Acc { common, io, wasm } = lines.join("\n");
+    let impl_import = if config.wasm_enabled {
```

Comment:
nit: Why are we importing differently when wasm is enabled, can we unify them?

Same comment for `common_import` below

In addition, I guess we already have a command line flag about whether to generate the impl and decl into two files or one file. So we may use that flag instead of whether wasm is enabled

Reply:
For now I decided that WASM requires the declaration to be separate to ease implementation, I might relax this requirement in the future however.